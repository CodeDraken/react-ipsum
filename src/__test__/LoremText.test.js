import React from 'react'
import { shallow } from 'enzyme'

import { LoremText } from '../'

describe('LoremText', () => {
  it('Should render a span tag by default', () => {
    const wrapper = shallow(<LoremText />)

    expect(wrapper.type()).toBe('span')
  })

  it('Should pass additional classnames', () => {
    const wrapper = shallow(<LoremText className='extra' />)

    expect(wrapper.hasClass('extra')).toBe(true)
  })

  it('Should render as other elements', () => {
    const wrapper = shallow(<LoremText renderAs='p' />)

    expect(wrapper.type()).toBe('p')
  })

  it('Should have a max length', () => {
    const wrapper = shallow(<LoremText len={10} />)

    expect(wrapper.text().length).toBe(10)
  })

  describe('Words', () => {
    it('Should render sub elements', () => {
      const wrapper = shallow(<LoremText unit='w' count={5} subEl='p' />)

      expect(wrapper.find('p').at(0).text()).toBeTruthy()
      expect(wrapper.find('p').length).toBe(5)
    })

    it('Should pass sub classes to sub elements', () => {
      const wrapper = shallow(<LoremText unit='w' count={5} subEl='p' subClass='custom' />)

      expect(wrapper.find('p').at(0).hasClass('custom')).toBe(true)
    })
  })

  describe('Sentences', () => {
    it('Should render sub elements', () => {
      const wrapper = shallow(<LoremText unit='s' count={5} subEl='p' />)

      expect(wrapper.find('p').at(0).text()).toBeTruthy()
      expect(wrapper.find('p').length).toBe(5)
    })

    it('Should pass sub classes to sub elements', () => {
      const wrapper = shallow(<LoremText unit='s' count={5} subEl='p' subClass='custom' />)

      expect(wrapper.find('p').at(0).hasClass('custom')).toBe(true)
    })
  })

  describe('Paragraphs', () => {
    it('Should render sub elements', () => {
      const wrapper = shallow(<LoremText unit='p' count={5} subEl='p' />)

      expect(wrapper.find('p').at(0).text()).toBeTruthy()
      expect(wrapper.find('p').length).toBe(5)
    })

    it('Should pass sub classes to sub elements', () => {
      const wrapper = shallow(<LoremText unit='p' count={5} subEl='p' subClass='custom' />)

      expect(wrapper.find('p').at(0).hasClass('custom')).toBe(true)
    })
  })

  describe('Snapshots', () => {
    it('Should exist', () => {
      expect(LoremText).toMatchSnapshot()
    })
  })
})
