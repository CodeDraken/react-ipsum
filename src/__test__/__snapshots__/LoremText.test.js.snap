// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LoremText Snapshots Should exist 1`] = `[Function]`;
